src/main/scala/scalaz/distributed/SupportedType.scala

```diff
@@ -0,0 +1,15 @@
+package scalaz.distributed
+
+private[distributed] trait SupportedType[A]
```

Comment:
We'll need to reify the type here, add some method like `reifiedType` that returns a sum type, of all possible types.